pkg/kp/lock.go

```diff
@@ -0,0 +1,100 @@
+package kp
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/square/p2/pkg/util"
+)
+
+type Lock struct {
+	client  *api.Client
+	session string
+	name    string
+}
+
+type AlreadyLockedError struct {
+	Key string
+}
+
+func (err AlreadyLockedError) Error() string {
+	return fmt.Sprintf("Key %q is already locked", err.Key)
+}
+
+func (s *Store) NewLock(name string) (Lock, error) {
```

Comment:
to make sure i'm understanding correctly, this does not ACTUALLY lock anything it just gives you a struct that you can then lock? i.e. gives you an "unlocked lock" back